python/cuml/tsa/arima.pyx

```diff
@@ -320,22 +322,16 @@ class ARIMA(Base):
         cdef uintptr_t d_sma_ptr = <uintptr_t> NULL
         cdef uintptr_t d_sigma2_ptr = <uintptr_t> NULL
         if order_kf.k:
-            d_mu, *_ = input_to_cuml_array(self.mu, check_dtype=np.float64)
-            d_mu_ptr = d_mu.ptr
+            d_mu_ptr = self._mu_.ptr
         if order_kf.p:
-            d_ar, *_ = input_to_cuml_array(self.ar, check_dtype=np.float64)
-            d_ar_ptr = d_ar.ptr
+            d_ar_ptr = self._ar_.ptr
         if order_kf.q:
-            d_ma, *_ = input_to_cuml_array(self.ma, check_dtype=np.float64)
-            d_ma_ptr = d_ma.ptr
+            d_ma_ptr = self._ma_.ptr
         if order_kf.P:
-            d_sar, *_ = input_to_cuml_array(self.sar, check_dtype=np.float64)
-            d_sar_ptr = d_sar.ptr
+            d_sar_ptr = self._sar_.ptr
         if order_kf.Q:
-            d_sma, *_ = input_to_cuml_array(self.sma, check_dtype=np.float64)
-            d_sma_ptr = d_sma.ptr
-        d_sigma2, *_ = input_to_cuml_array(self.sigma2, check_dtype=np.float64)
-        d_sigma2_ptr = d_sigma2.ptr
+            d_sma_ptr = self._sma_.ptr
+        d_sigma2_ptr = self._sigma2_.ptr
 
         cdef ARIMAParams[double] cpp_params
         cpp_params.mu = <double*> d_mu_ptr
```

Comment:
Building the `ARIMAParams` seems to happen several times (I think 4?) in this class. Might want to make an internal method that can be reused

Reply:
I don't think it can work with an internal method because `ARIMAParams` is a C++ type (as it's a template structure) so a Python function can't return it. What I went for after a bit of research is a wrapper Cython class around `ARIMAParams`. If you know a cleaner solution I'd be curious to hear about it.

Reply:
An internal method can return a C++ type if its defined as a `cdef` function and only accessed from cython code. However, I think the cython wrapper works well too.

Reply:
If I remember correctly, a non-`cdef` class can't have a `cdef` method, and the class here can't be `cdef`ed because it inherits from a Python class.